internal/discover/hooks.go

```diff
@@ -82,6 +82,7 @@ type Option func(*cdiHookCreator)
 type cdiHookCreator struct {
 	nvidiaCDIHookPath string
 	disabledHooks     map[HookName]bool
+	enableChmodHook   bool
```

Comment:
As mentioned in the addition of the flag. Does maintaining a map of "forced" hooks make more sense here than handling each hook separately?